src/gpu/intel/gemm/jit.hpp

```diff
@@ -74,17 +75,29 @@ struct gen_t : public primitive_t {
 
             // If we have both grouped scales and grouped zero-points, they must
             // have the same group size
-            if (a_scales_2d() && a_zp_2d()) {
+            if (a_scales_2d() && (a_zp_2d() || a_gs_2d())) {
+                gpu_assert(a_zp_2d() != a_gs_2d());
```

Comment:
If I'm not mistaken, with the new API, we can have any combination of a_scales_2d/a_zp_2d/a_gs_2d. So we can drop the gpu_assert here, right? Also checks should cover the case `a_zp_2d && a_gs_2d && !a_scales_2d`.

Reply:
Not sure, will check.